cellmlmanip/model.py

```diff
@@ -555,21 +555,51 @@ def get_symbol_by_ontology_term(self, namespace_uri, local_name):
             raise ValueError('Multiple variables annotated with {%s}%s' %
                              (namespace_uri, local_name))
 
-    def _get_symbols_by_rdf(self, predicate, object_=None):
+    def get_rdf_annotation(self, predicate):
```

Comment:
If you're only using this on model annotations, I'd add a subject to the original call. Even if it can be used more broadly, this can be optional. Perhaps call it `get_rdf_annotations(self, subject=None, predicate=None, object_=None)`?

There's potential for this to confuse if two subjects have the same local name; the docs should at least warn of this. Using full URIs would be safer though (and is what pycml does, so I'd rather we didn't regress!).

I'm not sure converting the RDF structure to a dict helps much, actually; to me it just seems to add complexity. Could this not just return the matching triples, and let the user decide what to do with them?

Reply:
It's true that it's a little complex and it does wrap annotation in a structure that still needs to be processed in general (although in my case I get exactly what I need).

Reason I did this was that returning the triples doesn't give me the annotation I have asked for and I'd have to start following the rdf graph to get to it. If we want this that's fine.


As an example I get the following triple:

`(rdflib.term.URIRef('#bondarenko_model_2004_apex'), rdflib.term.URIRef('https://chaste.comlab.ox.ac.uk/cellml/ns/pycml#named-attribute'), rdflib.term.BNode('N9d040fafd83243b4abbb9275d4eb8eba'))`

For this RDF:

```
   <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" >
      <rdf:Description rdf:about="#test_simple_odes">
          <named-attribute xmlns="https://chaste.comlab.ox.ac.uk/cellml/ns/pycml#">
             <rdf:Description>
                <name rdf:datatype="http://www.w3.org/2000/10/XMLSchema#string">SuggestedForwardEulerTimestep</name>
                <value rdf:datatype="http://www.w3.org/2000/10/XMLSchema#double">0.0002</value>
             </rdf:Description>
          </named-attribute>
   </rdf:RDF>
```